validation/genesis/genesis-allocs_test.go

```diff
@@ -184,6 +196,64 @@ func testGenesisAllocs(t *testing.T, chain *ChainConfig) {
 	require.Equal(t, string(expectedData), string(gotData))
 }
 
+// trim the CBOR octets from the bytecode of a weth9 contract
+func trimWeth9BytecodeMetadataHash(expectedData []byte, gotData []byte) ([]byte, []byte) {
+	var weth9ExpectedData, weth9GotData map[string]interface{}
+
+	err := json.Unmarshal(expectedData, &weth9ExpectedData)
+	if err != nil {
+		fmt.Printf("error parsing regenerated alloc: %s", err)
+	}
+
+	key := "0x4200000000000000000000000000000000000006"
+	if entry, ok := weth9ExpectedData[key].(map[string]interface{}); ok {
+		if code, ok := entry["code"].(string); ok {
+			// Trim the last 100 characters (50 bytes) which is the CBOR length in octets
+			if len(code) > 100 {
+				entry["code"] = code[:len(code)-100]
+			} else {
+				fmt.Println("The regenerated bytecode length is less than 100 characters; no trimming performed.")
+			}
+		} else {
+			log.Fatalf("Field 'code' not found or is not a string in regenerated bytecode: %s", key)
+		}
+	} else {
+		log.Fatalf("Key %s not found in regenerated bytecode", key)
+	}
+
+	// repeat for gotData
```

Comment:
Would it be neater to have this function just operate on a single `input []byte`, to reduce duplication? And then just call it twice (one for each regenerated/registry inputs).

Reply:
updated